core/src/main/java/io/questdb/log/AbstractLogRecord.java

```diff
@@ -376,7 +378,14 @@ public Utf8Sink putNonAscii(long lo, long hi) {
 
     @Override
     public LogRecord ts() {
-        sink().putISODate(clock.getTicks());
+        final long us = clock.getTicks();
+        if (LogLevel.TIMEZONE != null) {
+            final Instant instant = Instant.ofEpochMilli(us / 1000).plusNanos((us % 1000) * 1000);
```

Comment:
This will create a heap object of type Instace on every line logged. It's a no go for QuestDB